File: Data Structures & Algorithms/house-robber/submission-1.java
class Solution {
    int[] memo ;
    public int rob(int[] nums) {
        int n =nums.length;
        memo=new int[n];
        if(n<3){
            return Math.max(nums[0],nums[1]);

        }
        memo[0]=nums[0];
        memo[1]=nums[1];
        for (int i =2;i<n;i++){
            memo[i]=Math.max(nums[i]+memo[i-2],memo[i-1]);
        }
        return Math.max(memo[n-1],memo[n-2]);

        
    
        

    }
}

File: Data Structures & Algorithms/jump-game/submission-1.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        prev=-1
        i=0
        while i<len(nums)-1:
            print(f"prev= {prev}, i= {i}")
            prev=i       
            i=i+nums[i]
            if i==prev:
                return False

        if i==len(nums)-1 :
            return True
        else : 
            return False
        